packages/runtime/src/interpreter/interpreter.ts

```diff
@@ -67,12 +67,14 @@ export class Interpreter {
 	labelMap: Map<string, number>; // label string mapped to their respective indexes in instructions array
 	currentInnerTxnGroup: EncTx[]; // "current" inner transaction
 	innerTxnGroups: EncTx[][]; // executed inner transactions
-
+	cost: number; // total code
 	constructor() {
 		this.stack = new Stack<StackElem>();
+		this.mode = ExecutionMode.APPLICATION;
```

Comment:
why setting this? Interpreter can also run TEAL in `signature` mode, no?

Reply:
When we call [interpreter.execute](https://github.com/scale-it/algo-builder/blob/feat/global-OpcodeBudget/packages/runtime/src/interpreter/interpreter.ts#L485), we pass ExecuteMode to parameter. I just initial value at here.